src/views/CashlinkManage.vue

```diff
@@ -108,55 +112,56 @@ export default class CashlinkManage extends Vue {
+            this.isTxSent = false;
             this.retrievedCashlink = this.cashlink!;
         }
 
         if (!this.isTxSent) {
             // Note that this will never be called when coming from SignTransactionLedger as it sends and stores
             // the cashlink itself.
-            if (!NetworkClient.hasInstance()) {
-                NetworkClient.createInstance(Config.networkEndpoint);
-            }
-            const network = NetworkClient.Instance;
-            await network.init();
-
             if (!this.keyguardResult || !this.keyguardRequest) {
                 this.$rpc.reject(new Error('Unexpected: No valid Cashlink;'));
                 return;
             }
-            this.retrievedCashlink.networkClient = network;
-            network.on(NetworkClient.Events.API_READY,
+            network.$on(Network.Events.API_READY,
                 () => this.status = 'Contacting seed nodes...');
-            network.on(NetworkClient.Events.CONSENSUS_SYNCING,
+            network.$on(Network.Events.CONSENSUS_SYNCING,
                 () => this.status = 'Syncing consensus...');
-            network.on(NetworkClient.Events.CONSENSUS_ESTABLISHED,
+            network.$on(Network.Events.CONSENSUS_ESTABLISHED,
                 () => this.status = 'Sending transaction...');
-            network.on(NetworkClient.Events.TRANSACTION_PENDING,
+            network.$on(Network.Events.TRANSACTION_PENDING,
                 () => this.status = 'Awaiting receipt confirmation...');
+            this.retrievedCashlink.networkClient = await network.getNetworkClient();
 
             // Store cashlink in database first to be safe when browser crashes during sending
             await CashlinkStore.Instance.put(this.retrievedCashlink);
 
-            network.relayTransaction({
-                sender: new Nimiq.Address(this.keyguardRequest.sender).toUserFriendlyAddress(),
-                senderPubKey: this.keyguardResult.publicKey,
-                recipient:  new Nimiq.Address(this.keyguardRequest.recipient).toUserFriendlyAddress(),
-                value: Nimiq.Policy.lunasToCoins(this.keyguardRequest.value),
-                fee: Nimiq.Policy.lunasToCoins(this.keyguardRequest.fee),
+            transactionToSend = transactionToSend || await network.createTx({
+                sender: this.keyguardRequest.sender,
+                senderType: this.keyguardRequest.senderType,
+                recipient:  this.keyguardRequest.recipient,
+                recipientType: this.keyguardRequest.recipientType,
+                value: this.keyguardRequest.value,
+                fee: this.keyguardRequest.fee,
                 validityStartHeight: this.keyguardRequest.validityStartHeight,
+                data: this.keyguardRequest.data,
+                signerPubKey: this.keyguardResult.publicKey,
                 signature: this.keyguardResult.signature,
-                extraData: this.keyguardRequest.data,
-            });
```

Comment:
Seems like this whole block can be shortened with object spreading:

```js
transactionToSend = transactionToSend || await network.createTx({
    ...this.keyguardRequest,
    signerPubKey: this.keyguardResult.publicKey,
});

Reply:
:+1: